experimental/yarpl/include/reactivestreams/ReactiveStreams.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+#include <exception>
+#include <memory>
+
+namespace reactivestreams_yarpl {
+// TODO remove 'yarpl' if we agree on this usage elsewhere
+
+class Subscription {
+ public:
+  virtual ~Subscription() = default;
+  virtual void request(int64_t n) = 0;
+  virtual void cancel() = 0;
+};
+
+template <typename T>
+class Subscriber {
+ public:
+  virtual ~Subscriber() = default;
+  virtual void onSubscribe(Subscription*) = 0;
+  virtual void onNext(const T&) = 0;
```

Comment:
Should we assume that value is always movable, and will always be moved from? (We'll have a chain of onNext() calls down a pipeline.)